src/details/ArborX_DetailsSortUtils.hpp

```diff
@@ -175,9 +183,7 @@ sortObjects(Kokkos::Experimental::SYCL const &space, ViewType &view)
       oneapi::dpl::make_zip_iterator(view.data(), permute.data());
   oneapi::dpl::execution::device_policy policy(
       *space.impl_internal_space_instance()->m_queue);
-  oneapi::dpl::sort(
-      policy, zipped_begin, zipped_begin + n,
-      [](auto lhs, auto rhs) { return std::get<0>(lhs) < std::get<0>(rhs); });
+  oneapi::dpl::sort(policy, zipped_begin, zipped_begin + n, SYCLCompare{});
```

Comment:
Sounds more like a bug with the toolchain.  We do have `-fsycl-unnamed-lambda` in the compile line

Reply:
Possibly. I'm not sure sure what to think of intel/llvm#4769. Lets see if it fixes it. If not, I'll try to put the latest nightly and check.